codecs/opus/src/test/java/org/restcomm/media/codec/opus/OpusCodecTest.java

```diff
@@ -71,9 +79,36 @@ public void tearDown() throws Exception {
      */
     @Test
     public void testCodec() {
-    	OpusJni opus = new OpusJni();
-    	opus.setOpusObserverNative(this);
-    	opus.sayHelloNative();
+    	
+    	try {
+	        FileInputStream inputStream = new FileInputStream("src\\main\\jni\\test_sound_mono_48.pcm");
+	        FileOutputStream outputStream = new FileOutputStream("src\\main\\jni\\test_sound_mono_48_decoded.pcm", false);
+	    	
+	    	OpusJni opus = new OpusJni();
+	    	opus.setOpusObserverNative(this);
+	    	opus.sayHelloNative();
+	    	opus.initNative();
+
+	        try {
+	        	byte[] input = new byte[960];
+	        	short[] inputData = new short[480];
+        		byte[] output = new byte[960];
+	        	while (inputStream.read(input) == 960) {
+	        		ByteBuffer.wrap(input).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().get(inputData);
+	        		byte[] encodedData = opus.encodeNative(inputData);
+	        		short[] decodedData = opus.decodeNative(encodedData);
+	        		ByteBuffer.wrap(output).order(ByteOrder.LITTLE_ENDIAN).asShortBuffer().put(decodedData);
+	        		outputStream.write(output);
+	        	}
+	        } finally {
+	        	inputStream.close();
+	        	outputStream.close();
+	        	opus.closeNative();
+	        }
+    	} catch (IOException exc) {
+    		System.out.println("IOException: " + exc.getMessage());
+        	return;
```

Comment:
Fail the test instead of silently ignoring the error.

Reply:
Fixed